src/CMakeLists.txt

```diff
@@ -18,10 +18,25 @@ set(libsrc
   zos-tls.cc
   zos.cc
   zos-mount.c
-  celquopt.s
 )
 
-add_library(libzoslib OBJECT ${libsrc})
+set(CELQUOPT_OBJECT "${CMAKE_CURRENT_BINARY_DIR}/celquopt.s.o")
+set(CELQUOPT_SOURCE "${CMAKE_CURRENT_SOURCE_DIR}/celquopt.s")
+set_source_files_properties(${CELQUOPT_OBJECT}
+  PROPERTIES EXTERNAL_OBJECT TRUE GENERATED TRUE
+)
+add_custom_command(OUTPUT ${CELQUOPT_OBJECT}
+COMMAND /bin/as -mgoff -o ${CELQUOPT_OBJECT} ${CELQUOPT_SOURCE}
```

Comment:
Is this a workaround because of a cmake limitation or clang limitation?

Reply:
I think both.  cmake sends .s files to c/c++ compiler.  That might be fine if the -fno-integrated-as was also passed.  And clang can't compile .s files without that option.  We can't make the option the default since it impacts the asm() statements in code the wrong way.